### PR TITLE
Tile: add back old api

### PR DESCRIPTION
**Pull request by gretanausedaite**
## Changes

Adding back props API. Now Tile works both ways:

```js
<Tile.Wrapper>
  <Tile.Name name='Best tile ever' />
  <Tile.ThumbnailArea>
    <Tile.ThumbnailPicture url='/local-picture' />
  </Tile.ThumbnailArea>
</Tile.Wrapper>
```

```js
<Tile
  name='Second best tile ever'
  thumbnail='/another-local-file'
/>
```

## Testing

Added unit test

## Docs

Changeset

### CODE REVIEW

File: packages/itwinui-react/src/core/Tile/Tile.test.tsx [hunk 1]
```diff
@@ -273,45 +273,121 @@ it('should propagate misc props correctly', () => {
...
+      buttons={[
+        <Button key='button-1'>Button 1</Button>,
+        <Button key='button-2'>Button 2</Button>,
+      ]}
+      moreOptions={[
+        <MenuItem key='menuitem-1'>Item 1</MenuItem>,
+        <MenuItem key='menuitem-1'>Item 2</MenuItem>,
+      ]}
+      leftIcon={
+        <Tile.IconButton>
```
Comment (mayank99) [inline packages/itwinui-react/src/core/Tile/Tile.test.tsx:337]: why `IconButton` doesn't work here?

Comment (gretanausedaite) [reply packages/itwinui-react/src/core/Tile/Tile.test.tsx:337]: It works. Users must pass size='small' button for it to be "as designed" 
left is Tile.IconButton, right is IconButton:
image

+ off topic:
I think we should fix dark icons within those buttons. CSS tests have them white.

Comment (mayank99) [reply packages/itwinui-react/src/core/Tile/Tile.test.tsx:337]: oh wow, the icons are messed up only in light theme so i didn't even notice 😅

File: packages/itwinui-react/src/core/Tile/Tile.tsx [hunk 2]
```diff
@@ -397,10 +397,187 @@ TileMoreOptions.displayName = 'Tile.MoreOptions';
...
+  /**
+   * Flag whether the tile is disabled.
+   *
+   * Note: This only affects the tile. You need to manually disable
+   * the buttons and other interactive elements inside the tile.
+   *
+   * @default false
+   */
+  isDisabled?: boolean;
+  onClick?: React.MouseEventHandler<HTMLElement>;
```
Comment (mayank99) [inline packages/itwinui-react/src/core/Tile/Tile.tsx:503]: add jsdoc here? it's not obvious that this onClick is not applied on the wrapper

File: packages/itwinui-react/src/core/Tile/Tile.tsx [hunk 3]
```diff
@@ -410,23 +587,40 @@ TileButtons.displayName = 'Tile.Buttons';
...
+ *  rightIcon={<IconButton><SvgStar /></IconButton>}
+ *  isSelected={true}
+ *  isNew={false}
+ * />
  */
 export const Tile = Object.assign(TileComponent, {
+  /**
+   *
+   */
+  Wrapper: TileWrapper,
```
Comment (mayank99) [inline packages/itwinui-react/src/core/Tile/Tile.tsx:618]: fill jsdocs?

File: packages/itwinui-react/src/core/Tile/Tile.tsx [hunk 4]
```diff
@@ -397,10 +397,187 @@ TileMoreOptions.displayName = 'Tile.MoreOptions';
...
+  isSelected?: boolean;
+  /**
+   * Whether the tile is "new". Tile name becomes bold and gets a new status icon.
+   */
+  isNew?: boolean;
+  /**
+   * Default tile variant or the folder layout.
+   * @default 'default'
+   */
+  variant?: 'default' | 'folder';
```
Comment (mayank99) [inline packages/itwinui-react/src/core/Tile/Tile.tsx:479]: does folder tile work correctly? i think we had some issues supporting both variants with old apis

File: packages/itwinui-react/src/core/Tile/Tile.test.tsx (hunk 1 above)
Comment (mayank99) [reply packages/itwinui-react/src/core/Tile/Tile.test.tsx:337]: for now i would say use `IconButton` with manual `size='small'` in this unit test. we don't want to mix and match old and new apis.

the color issue can be fixed in a separate PR

File: packages/itwinui-react/src/core/Tile/Tile.tsx (hunk 4 above)
Comment (mayank99) [reply packages/itwinui-react/src/core/Tile/Tile.tsx:479]: just tested in the playground and it's working correctly, even with `onClick`.

looks like it was fixed in https://github.com/iTwin/iTwinUI/pull/1255#issuecomment-1553662141 + https://github.com/iTwin/iTwinUI/pull/1255#issuecomment-1561251594

**Review by mayank99 — APPROVED** [review]

**Review by rohan-kadkol — COMMENTED** [review]
Would be good if there are some examples using the old Tile API. (Similar to https://github.com/iTwin/iTwinUI/pull/1502#pullrequestreview-1587639586)

`examples/Tile.main.tsx` and the other Tile examples that are not currently in the PR are giving type errors. I believe `Tile` should be changed to `Tile.Wrapper` for the other Tile examples where we want to demo the new API.

image

**Comment by mayank99** [comment]
i've updated the missed references. i'll leave it on Greta to handle the rest

**Comment by rohan-kadkol** [comment]
> Would be good if there are some examples using the old Tile API. (Similar to https://github.com/iTwin/iTwinUI/pull/1502#pullrequestreview-1587639586)

@gretanausedaite I noticed you added this to the merge queue. Are we not adding examples using the old Tile API to the website?

**Comment by gretanausedaite** [comment]
> @gretanausedaite I noticed you added this to the merge queue. Are we not adding examples using the old Tile API to the website?

Created #1538

**Pull request merged**